### PR TITLE
enable api keys for query endpoints

### PR DESCRIPTION
**Pull request by solomon-negusse**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Make sure you are requesting to pull a topic/feature/bugfix branch (right side). Don't request your master!
- [x] Make sure you are making a pull request against the develop branch (left side). Also you should start your branch off our develop.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

query endpoints are not protected with api key

## What is the new behavior?

query endpoints protected and rate limits enabled..

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: terraform/modules/api_gateway/gateway/variables.tf [hunk 1]
```diff
@@ -52,13 +52,13 @@ variable "api_gateway_usage_plans" {
   default = {
     internal_apps = {
       quota_limit = 10000 # per day
-      burst_limit = 100   # per second
-      rate_limit  = 200
+      burst_limit = 300
```
Comment (danscales) [inline terraform/modules/api_gateway/gateway/variables.tf:55]: Maybe increase quota_limit and burst_limit even more (50000/1000)?  Kinshuk said we need high limits for processing of large lists - there could be a lot of requests if a customer creates a new large lists with 25000 locations.

File: tests_v2/unit/app/routes/datasets/test_query.py [hunk 2]
```diff
@@ -268,12 +268,17 @@ async def test_query_dataset_raster_geostore_huge(
...
 async def test_query_vector_asset_disallowed_1(
-    generic_vector_source_version, async_client: AsyncClient
+    generic_vector_source_version, apikey, async_client: AsyncClient
 ):
     dataset, version, _ = generic_vector_source_version
 
+    api_key, payload = apikey
+    origin = "https://" + payload["domains"][0]
+    headers = {"origin": origin, "x-api-key": api_key}
+
```
Comment (danscales) [inline tests_v2/unit/app/routes/datasets/test_query.py:278]: Would be nice to make a single function get_header that replaces these three lines everywhere with `headers = get_header(apikey)`

File: terraform/modules/api_gateway/gateway/main.tf [hunk 3]
```diff
@@ -73,7 +73,7 @@ module "query_post" {
   authorizer_id = aws_api_gateway_authorizer.api_key.id
   api_resource  = module.query_resource.aws_api_gateway_resource
 
-  require_api_key = false
+  require_api_key = true
```
Comment (danscales) [inline terraform/modules/api_gateway/gateway/main.tf:76]: I don't see a terraform module here for zonal statistics, similar to these 'query_post' and 'query_get' modules.  Do we need a terraform module for zonal statistics as well to actually enable rate limiting for zona statistics?

**Review by danscales — APPROVED** [review]
Looks great!  Thanks for doing this!

File: terraform/modules/api_gateway/gateway/main.tf (hunk 3 above)
Comment (solomon-negusse) [reply terraform/modules/api_gateway/gateway/main.tf:76]: Good catch, that's a deprecated endpoint and not used much looking at the usage stats. How about I merge this and address that in future iteration.

File: terraform/modules/api_gateway/gateway/variables.tf (hunk 1 above)
Comment (solomon-negusse) [reply terraform/modules/api_gateway/gateway/variables.tf:55]: @danscales: adding this to develop branch, merged before adding.

**Pull request merged**